File: vital-core-processor/src/main/java/me/vitalframework/processor/VitalPluginInfoHolder.java
package me.vitalframework.processor;

import lombok.NonNull;

/**
 * Generic Interface responsible for holding the contents for automatic `plugin.yml` creation between multiple Annotation Processor modules.
 *
 * @author xRa1ny
 */
public interface VitalPluginInfoHolder {
    /**
     * Cache variable storing information for later plugin.yml generation during compile-time.
     */
    @NonNull
    StringBuilder PLUGIN_INFO = new StringBuilder();
}

File: vital-core-processor/src/main/kotlin/me/vitalframework/processor/VitalPluginInfoHolder.kt
package me.vitalframework.processor

/**
 * Generic Interface responsible for holding the contents for automatic `plugin.yml` creation between multiple Annotation Processor modules.
 */
object VitalPluginInfoHolder {
    /**
     * Cache variable storing information for later plugin.yml generation during compile-time.
     */
    @JvmField
    val PLUGIN_INFO: StringBuilder = StringBuilder()
}